services/audit-service/src/main/java/datawave/microservice/audit/replay/ReplayController.java

```diff
@@ -0,0 +1,726 @@
+            
+            // if we own it, stop it. otherwise, fire an event to all of the audit services
+            RunningReplay replay = runningReplays.get(status.getId());
+            if (replay != null) {
+                replay.getStatus().setState(STOPPED);
+                statusCache.update(replay.getStatus());
+                
+                try {
+                    replay.getFuture().get(replayProperties.getStopGracePeriodMillis(), TimeUnit.MILLISECONDS);
+                } catch (Exception e) {
+                    // interrupts and timeouts are ok. we just want to give some time to cleanup.
+                }
+                
+                // if it's still not done, cancel it
+                if (!replay.getFuture().isDone())
+                    replay.getFuture().cancel(true);
+                
+                runningReplays.remove(status.getId());
+            } else if (publishEvent) {
+                appCtx.publishEvent(new AuditReplayRemoteRequestEvent(this, busProperties.getId(), Request.stop(status.getId())));
+            }
+        } else {
+            return false;
+        }
+        
+        return true;
+    }
+    
+    /**
+     * Stops all audit replays
+     *
+     * @return status, indicating the number of audit replays which were successfully stopped
+     */
+    @ApiOperation(value = "Stops all audit replays.")
+    @RolesAllowed({"Administrator", "JBossAdministrator"})
+    @RequestMapping(path = "/stopAll", method = org.springframework.web.bind.annotation.RequestMethod.POST)
+    public String stopAll() {
+        
+        log.info("Stopping all audit replays");
+        
+        int replaysStopped = stopAll(replayProperties.isPublishEvents());
+        
+        String resp = replaysStopped + " audit replays stopped";
+        log.info(resp);
+        return resp;
+    }
+    
+    private int stopAll(boolean publishEnabled) {
+        int replaysStopped = 0;
+        
+        if (publishEnabled)
+            appCtx.publishEvent(new AuditReplayRemoteRequestEvent(this, busProperties.getId(), Request.stopAll()));
+        
+        // stop all of our replays. then, send an event out to all audit services to stop all replays
+        List<Status> statuses = statusAll().stream().filter(status -> status.getState() == RUNNING).collect(Collectors.toList());
+        for (Status status : statuses) {
+            if (stop(status, false))
+                replaysStopped++;
+        }
+        
+        return replaysStopped;
+    }
+    
+    /**
+     * Resumes an audit replay
+     *
+     * @param id
+     *            The audit replay id
+     * @return status, indicating whether the audit replay was successfully resumed
+     */
+    @ApiOperation(value = "Resumes an audit replay.")
+    @RolesAllowed({"Administrator", "JBossAdministrator"})
+    @RequestMapping(path = "/{id}/resume", method = org.springframework.web.bind.annotation.RequestMethod.POST)
```

Comment:
The "PUT" verb is usually used for update/modification of a resource.

Reply:
Updated in the next commit.